test/data/keymaps/empty-symbol-decl.xkb

```diff
@@ -0,0 +1,8 @@
+xkb_keymap {
+    xkb_keycodes  { include "evdev+aliases(qwerty)" };
+    xkb_types     { include "complete" };
+    xkb_compat    { include "complete" };
+    xkb_symbols "sym" {
+        key <SPC> { , [Space] };
```

Comment:
```suggestion
        // This dubious syntax is accepted by xkbcomp so we accept it for
        // backward compatibility. The preferred way to write it is:
        // key <SPC> { [], [Space] };
        key <SPC> { , [Space] };
```

Reply:
Actually, these syntaxes are not equivalent. Using X11 `xkbcomp`, I get that 
```c
key <SPC> { [Space] };
key <SPC> { , [exclam] };
```
is equivalent to 
```c
key <SPC> { [exclam] };
```
whereas  
```c
key <SPC> { [Space] };
key <SPC> { [], [exclam] };
```
is equivalent to 
```c
key  <SPC> {
    symbols[Group1]= [          Space ],
    symbols[Group2]= [          exclam ]
};
```
That’s why I wrote it was probably an oversight.

Reply:
Thanks for the clarification. then, is there any other changes needed?

Reply:
@chiro  I wonder: where did you encounter this syntax in the first place? Unless there is a real-world proof, I think we should fix just the segfault, but disallow this syntax. It is useless per se and confusing. It is not used in `xkeyboard-config`.

I do not see to make `key <SPC> { , [exclam] };` equivalent to `key <SPC> { [], [exclam] };` either: XKB has C-style, so it would not be intuitive.

Reply:
Fully agree with @wismill here - if this is somewhere in the xkb maps we should fix it in xkeyboard-config but allowing this syntax seems like the wrong approach.

Reply:
@wismill It was not found in an actual in-use layout file. It's detected in testing. As far as I know, there is no real-world use case of this syntax. I have no objection to disallowing this syntax.

Reply:
> @wismill It was not found in an actual in-use layout file. It's detected in testing. As far as I know, there is no real-world use case of this syntax. I have no objection to disallowing this syntax.

Just to clarify, I couldn't find any real-world use case, but it doesn't mean there is no use case in the world. That's why I made the CL just to fix the crash.

Reply:
@chiro we are grateful for uncovering the segfault. This is a nasty bug, and it would not occur to me to test this syntax. Well done!

I am curious, could you share how you stumble on this? Maybe we could then improve our test suite further.

Now, given that:
- AFAIK the syntax is useless and undocumented;
- `xkeyboard-config` compiles just fine with xkbcommon;
- no one opened a ticket before.

then I really doubt there are use cases in the world. While we strive to have maximum compatibility with X11, I think this is one _edge_ case we should not support. Probably worth a comment, at least in the code.

Reply:
It was reported as a potential security issue to the project I'm working on (Actually it was not because only layout files in xkeyboard-config are used). I don't know the details of how it was discovered, but it should be by fuzzing.

Reply:
If it's not used in any existing xkeyboard-config files, can we just make it return a parser error? That seems like the best option here - making this intentionally backwards-compatible means we have to support it forever when there seems to be little point to it.

Reply:
Thanks for the comment (and the reaction)! I changed this PR not to allow the syntax. Could you review it again? Thanks.